pkg/controller/storagecluster/cephcluster.go

```diff
@@ -403,6 +403,10 @@ func newStorageClassDeviceSets(sc *ocsv1.StorageCluster, serverVersion *version.
 						in := getPlacement(sc, "osd-prepare-tsc")
 						(&in).DeepCopyInto(&preparePlacement)
 
+						if sc.Spec.FlexibleScaling == true {
```

Comment:
Please compare against topologyKey instead of FlexibleScaling because the storageDeviceSets also has a topologyKey.

Reply:
Thanks, comparing against the topology key seems to be appropriate and changes are done!